cmake/linker/ld/target_base.cmake

```diff
@@ -4,6 +4,10 @@
 
 macro(toolchain_ld_base)
 
+  if(NOT PROPERTY_LINKER_SCRIPT_DEFINES)
```

Comment:
Moving this can change whether this if-statement is evaluated to true or false.

Can you prove somehow that this is safe?

Reply:
At least grepping indicates that this is safe:
- The only other reference of `PROPERTY_LINKER_SCRIPT_DEFINES` is from `zephyr/arch/x86/CMakeLists.txt` which will be included resulting from `add_subdirectory(arch)`.
- This makes this PR equivalent to moving the `-D__GCC_LINKER_CMD__` down from line 46 to line 227, which retains the same relative order as before.

```
$ rg 'PROPERTY_LINKER_SCRIPT_DEFINES|toolchain_ld_base|add_subdirectory\(arch'
zephyr/CMakeLists.txt
45:if(NOT PROPERTY_LINKER_SCRIPT_DEFINES)
46:  set_property(GLOBAL PROPERTY PROPERTY_LINKER_SCRIPT_DEFINES -D__GCC_LINKER_CMD__)
227:toolchain_ld_base()
465:add_subdirectory(arch)

zephyr/arch/x86/CMakeLists.txt
13:  set_property(GLOBAL APPEND PROPERTY PROPERTY_LINKER_SCRIPT_DEFINES -D__IAMCU)
```